gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcWriteChannel.java

```diff
@@ -111,33 +112,39 @@ protected String getResourceString() {
   }
 
   @Override
-  public void handleResponse(Object response) {
+  public void handleResponse(WriteObjectResponse response) {
+    Object resource = response.getResource();
     Map<String, byte[]> metadata =
-        response.getMetadataMap().entrySet().stream()
+        resource.getMetadataMap().entrySet().stream()
             .collect(
                 toMap(Map.Entry::getKey, entry -> BaseEncoding.base64().decode(entry.getValue())));
 
-    byte[] md5Hash =
-        !response.getMd5Hash().isEmpty()
-            ? BaseEncoding.base64().decode(response.getMd5Hash())
-            : null;
+    byte[] md5Hash = null;
+    byte[] crc32c = null;
 
-    byte[] crc32c =
-        response.hasCrc32C()
-            ? ByteBuffer.allocate(4).putInt(response.getCrc32C().getValue()).array()
-            : null;
+    if (resource.hasChecksums()) {
+      md5Hash =
+          !resource.getChecksums().getMd5Hash().isEmpty()
```

Comment:
Should we use `hasMd5Hash()` in a check?

Reply:
`hasMd5HasH` is not a supported API, hence the check `getMd5Hash().isEmpty()`.